octopusdeploy_framework/resource_team_mappers.go

```diff
@@ -0,0 +1,249 @@
+package octopusdeploy_framework
+
+import (
+	"context"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/client"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/core"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/teams"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/userroles"
+	"github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy_framework/schemas"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func mapTeamStateToResource(ctx context.Context, model schemas.TeamModel) *teams.Team {
+	name := model.Name.ValueString()
+	team := teams.NewTeam(name)
+	team.ID = model.ID.ValueString()
+
+	team.Description = model.Description.ValueString()
+	team.SpaceID = model.SpaceID.ValueString()
+	team.MemberUserIDs = expandStringSet(model.Users)
+
+	if !model.ExternalSecurityGroup.IsNull() && !model.ExternalSecurityGroup.IsUnknown() {
+		team.ExternalSecurityGroups = mapExternalSecurityGroupsStateToResource(ctx, model.ExternalSecurityGroup)
+	} else {
+		team.ExternalSecurityGroups = []core.NamedReferenceItem{}
+	}
+
+	if !model.CanBeDeleted.IsNull() && !model.CanBeDeleted.IsUnknown() {
```

Comment:
These if null or unknown checks aren't strictly necessary, but should protect us from unexpected errors related to inconsistent or unknown state.